cmd/tools/gendynamicconfig/main.go

```diff
@@ -0,0 +1,245 @@
+		},
+		{
+			Name:       "Namespace",
+			GoArgs:     "(namespace string)",
+			GoArgNames: "(namespace)",
+		},
+		{
+			Name:       "NamespaceID",
+			GoArgs:     "(namespaceID string)",
+			GoArgNames: "(namespaceID)",
+		},
+		{
+			Name:       "TaskQueue",
+			GoArgs:     "(namespace string, taskQueue string, taskQueueType enumspb.TaskQueueType)",
+			GoArgNames: "(namespace, taskQueue, taskQueueType)",
+		},
+		{
+			Name:       "ShardID",
+			GoArgs:     "(shardID int32)",
+			GoArgNames: "(shardID)",
+		},
+		{
+			Name:       "TaskType",
+			GoArgs:     "(taskType enumsspb.TaskType)",
+			GoArgNames: "(taskType)",
+		},
+	}
+)
+
+func panicIfErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func writeTemplatedCode(w io.Writer, text string, data any) {
+	panicIfErr(template.Must(template.New("code").Parse(text)).Execute(w, data))
+}
+
+func generateTypeEnum(w io.Writer, tp *settingType) {
+	writeTemplatedCode(w, `
+const Type{{.Name}} Type = {{.Index}} // go type: {{.GoType}}
+`, tp)
+}
+
+func generatePrecEnum(w io.Writer, prec *settingPrecedence) {
+	writeTemplatedCode(w, `
+const Precedence{{.Name}} Precedence = {{.Index}}
+`, prec)
+}
+
+func generateType(w io.Writer, tp *settingType, prec *settingPrecedence) {
+	writeTemplatedCode(w, `
+type {{.T.Name}}{{.P.Name}}Setting Setting[{{.T.GoType}}, func{{.P.GoArgs}}]
+
+func New{{.T.Name}}{{.P.Name}}Setting(key Key, def {{.T.GoType}}, description string) {{.T.Name}}{{.P.Name}}Setting {
+	s := {{.T.Name}}{{.P.Name}}Setting{
+		key:         key,
+		def:         def,
+		description: description,
+	}
+	return s
+}
+
+func New{{.T.Name}}{{.P.Name}}SettingWithConstrainedDefault(key Key, cdef []TypedConstrainedValue[{{.T.GoType}}], description string) {{.T.Name}}{{.P.Name}}Setting {
+	s := {{.T.Name}}{{.P.Name}}Setting{
+		key:         key,
+		cdef:        cdef,
+		description: description,
+	}
+	return s
+}
+
+func (s {{.T.Name}}{{.P.Name}}Setting) Key() Key               { return s.key }
+func (s {{.T.Name}}{{.P.Name}}Setting) Type() Type             { return Type{{.T.Name}} }
+func (s {{.T.Name}}{{.P.Name}}Setting) Precedence() Precedence { return Precedence{{.P.Name}} }
+
+func (s {{.T.Name}}{{.P.Name}}Setting) WithDefault(v {{.T.GoType}}) {{.T.Name}}{{.P.Name}}Setting {
+	newS := s
+	newS.def = v
+	return newS
+}
+
+type {{.T.Name}}PropertyFn{{if .NotGlobal}}With{{.P.Name}}Filter{{end}} func{{.P.GoArgs}} {{.T.GoType}}
+
+func (c *Collection) Get{{.T.Name}}{{if .NotGlobal}}By{{.P.Name}}{{end}}(s {{.T.Name}}{{.P.Name}}Setting) {{.T.Name}}PropertyFn{{if .NotGlobal}}With{{.P.Name}}Filter{{end}} {
+	return func{{.P.GoArgs}} {{.T.GoType}} {
+		return matchAndConvert(
+			c,
+			(Setting[{{.T.GoType}}, func{{.P.GoArgs}}])(s),
+			precedence{{.P.Name}}{{.P.GoArgNames}},
+			convert{{.T.Name}},
+		)
+	}
+}
+
+func Get{{.T.Name}}PropertyFn{{if .NotGlobal}}FilteredBy{{.P.Name}}{{end}}(value {{.T.GoType}}) {{.T.Name}}PropertyFn{{if .NotGlobal}}With{{.P.Name}}Filter{{end}} {
```

Comment:
Do we really need the `Get` prefix for these calls? It could just be `dynamicconfig.Duration(bar)`

Reply:
If we swap the order (discussed above) we have only the Get instead of the type. Maybe that's better?

Reply:
I think it is, yeah. I forgot I made the other recommendation :P